### PR TITLE
Check if resources implement scale in disruption controller

### PR DESCRIPTION
**Pull request by mortent**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
PDBs will not work in most cases if the selector includes pods that does not implement the scale subresource as shown in #77383. This fix makes sure we can distinguish between a pod's owner ref pointing to a non-existent resource and it pointing to a resource that does not implement the scale subresource. The error will be included in the warning event generated by the controller and will help users understand the cause of the problem.

We only look up scale if `maxUnavailable` is specified, or `minAvailable` is specified as a percentage.

**Which issue(s) this PR fixes**:
Fixes #

**Special notes for your reviewer**:
This is the same PR as https://github.com/kubernetes/kubernetes/pull/86828 which got closed by the fejta-bot.

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

/sig apps
@kubernetes/sig-apps-pr-reviews

### CODE REVIEW

**Review by janetkuo — APPROVED** [review]

File: cmd/kube-controller-manager/app/policy.go [hunk 1]
```diff
@@ -67,6 +67,7 @@ func startDisruptionController(ctx ControllerContext) (http.Handler, bool, error
 		client,
 		ctx.RESTMapper,
 		scaleClient,
+		client.Discovery(),
```
Comment (cheftako) [inline cmd/kube-controller-manager/app/policy.go:70]: We've had problems in the past with discovery error handling in controllers causing the KCM/CCM to crash loop. Can you look at #73337 and add the disruption controller to the list of controllers we check for problems with discovery on init?

**Comment by mortent** [comment]
@cheftako I have added the disruption controller to the list in the TestController_DiscoveryError test.

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**